components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -353,25 +353,59 @@ func isSuccess(status int) bool {
 	return false
 }
 
-func isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
-	if runtime.Status.Provisioning != nil &&
-		runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
-		runtime.Status.Deprovisioning == nil {
+// Helper function which returns true if the runtime state is trackable
+func isTrackableState(state kebruntime.State) bool {
+	switch state {
+	case kebruntime.StateSucceeded, kebruntime.StateError, kebruntime.StateUpgrading, kebruntime.StateUpdating:
 		return true
 	}
 	return false
 }
 
+func (p *Process) isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
```

Comment:
Does it make sense to simplify the function logic to do something similar to the following pseudo-code?

```golang
func (p *Process) isRuntimeTrackable(runtime *kebruntime.RuntimeDTO) bool {
	// log some stuff

	return isTrackableState(runtime.Status.State) || isProvisioned(runtime)
}
```

Reply:
Done